components/plan-build/bin/composite_build_functions.bash

```diff
@@ -141,28 +142,29 @@ _validate_bind_mappings() {
 
     if [[ -f "${resolved}/BINDS" ]]; then
       while read -r line; do
-        IFS== read bind_name exports <<< "${line}"
-        all_binds_for_pkg[$bind_name]="${exports[@]}"
+        IFS="=" read -r bind_name exports <<< "${line}"
+        all_binds_for_pkg[$bind_name]="${exports[*]}"
       done < <(_read_metadata_file_for "${resolved}" BINDS)
     fi
     if [[ -f "${resolved}/BINDS_OPTIONAL" ]]; then
       while read -r line; do
-        IFS== read bind_name exports <<< "${line}"
+        IFS="=" read -r bind_name exports <<< "${line}"
         if [[ -n "${all_binds_for_pkg[$bind_name]}" ]]; then
           exit_with "The bind ${bind_name} has already been declared in pkg_binds for package ${pkg}, it cannot also be declared in pkg_binds_optional"
         fi
-        all_binds_for_pkg[$bind_name]="${exports[@]}"
+        all_binds_for_pkg[$bind_name]="${exports[*]}"
       done < <(_read_metadata_file_for "${resolved}" BINDS_OPTIONAL)
     fi
 
     unset bind_mappings
     bind_mappings=("${pkg_bind_map[$pkg]}")
 
     # This is space-delimited, so no quotes
+    # shellcheck disable=2068,2180
```

Comment:
```
In ./components/plan-build/bin/composite_build_functions.bash line 163:
    for mapping in ${bind_mappings[@]}; do
                   ^-- SC2068: Double quote array expansions to avoid re-splitting elements.
```
I think this should be
```
    for mapping in "${bind_mappings[@]}"; do
```
but I wanted to double-check since there's some strange stuff going on at line 175 and 181:

```
In ./components/plan-build/bin/composite_build_functions.bash line 175:
      satisfying_package_exports=("${pkg_export_map[$resolved_satisfying_package][@]}")
                                   ^-- SC2180: Bash does not support multidimensional arrays. Use 1D or associative arrays.
```

```
In ./components/plan-build/bin/composite_build_functions.bash line 181:
      for required_exported_value in ${all_binds_for_pkg[$bind_name][@]}; do
                                     ^-- SC2068: Double quote array expansions to avoid re-splitting elements.
                                     ^-- SC2180: Bash does not support multidimensional arrays. Use 1D or associative arrays.
```